src/nvim/strings.c

```diff
@@ -1201,45 +1217,63 @@ int vim_vsnprintf(char *str, size_t str_m, const char *fmt, va_list ap,
             if (fmt_spec == 'g' || fmt_spec == 'G') {
               // can't use %g directly, cause it prints "1.0" as "1"
               if ((abs_f >= 0.001 && abs_f < 10000000.0) || abs_f == 0.0) {
-                fmt_spec = 'f';
+                fmt_spec = ASCII_ISUPPER(fmt_spec) ? 'F' : 'f';
               } else {
                 fmt_spec = fmt_spec == 'g' ? 'e' : 'E';
               }
               remove_trailing_zeroes = true;
             }
 
-            if (fmt_spec == 'f' && abs_f > 1.0e307) {
-              // avoid a buffer overflow
-              memmove(tmp, "inf", sizeof("inf"));
-              str_arg_l = sizeof("inf") - 1;
+            if ((fmt_spec == 'f' || fmt_spec == 'F') && abs_f > 1.0e307) {
```

Comment:
I think that this should be `isinf(f) || ((fmt_spec == 'f' || fmt_spec == 'F') && abs_f > 1.0e307)`. And `isinf(f)` branch below could be removed then.